diag_manager/fms_diag_file_object.F90

```diff
@@ -1005,18 +1009,18 @@ subroutine add_start_time(this, start_time)
   class(fmsDiagFile_type), intent(inout)       :: this           !< The file object
   TYPE(time_type),         intent(in)          :: start_time     !< Start time to add to the fileobj
 
-  !< If the start_time sent in is equal to the base_time return because
-  !! this%start_time was already set to the base_time
-  if (start_time .eq. get_base_time()) return
+  !< If the start_time sent in is equal to the diag_init_time return because
+  !! this%start_time was already set to the diag_init_time
+  if (start_time .eq. diag_init_time) return
 
-  if (this%start_time .ne. get_base_time()) then
-    !> If the this%start_time is not equal to the base_time from the diag_table
+  if (this%start_time .ne. diag_init_time) then
+    !> If the this%start_time is not equal to the diag_init_time from the diag_table
```

Comment:
I only see a test where the date is prepended correctly.  Is there a way to force this scenario to ensure it is caught?

Reply:
The second test in [test_prepend_date.sh](https://github.com/NOAA-GFDL/FMS/pull/1540/files#diff-f98593e6e7ff4d938f937f53d8ed60454134e7fc8f3b04ff12845c3dedd5b664) should catch this.